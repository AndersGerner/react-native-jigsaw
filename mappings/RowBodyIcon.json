[{"name":"Row Single Line Body Icon","tag":"RowBodyIcon","description":"A row with left aligned body text and a right aligned icon","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_SingleLine_BodyIcon.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"icon":{"label":"Icon","description":"Icon to display","type":"icon","value":null,"editable":true}},"layout":{}},{"name":"Row Double Line Body Icon","tag":"RowBodyIcon","description":"A row with left aligned body text and subtitle text and a right aligned icon","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_DoubleLine_BodyIcon.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"subtitle":{"label":"Subtitle","description":"Subtitle text to display","type":"string","value":"San Diego","editable":true},"icon":{"label":"Icon","description":"Icon to display","type":"icon","value":null,"editable":true}},"layout":{}}]