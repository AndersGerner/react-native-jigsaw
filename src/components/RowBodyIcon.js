/* @flow */

import * as React from "react";
import { View, Text, Image, StyleSheet } from "react-native";
import { withTheme } from "../core/theming";
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types";
import Row from "./Row";
import Icon from "./Icon";
import Config from "./Config";
import type { Theme } from "../types";

type Props = {
  title: string,
  subtitle?: string,
  icon: string,
  style?: any,
  theme: Theme
};

class RowBodyIcon extends React.Component<Props> {
  render() {
    const {
      title,
      subtitle,
      icon,
      style,
      theme: { colors, typography, spacing }
    } = this.props;

    return (
      <Row
        titleTypeStyle={typography.body1}
        titleColor={colors.medium}
        subtitleTypeStyle={typography.subtitle2}
        subtitleColor={colors.light}
        title={title}
        subtitle={subtitle}
        right={() => (
          <Icon
            name={icon}
            size={Config.rowSingleLineIconSize}
            color={colors.light}
            style={{ marginLeft: spacing.large }}
          />
        )}
        style={style}
      />
    );
  }
}

export default withTheme(RowBodyIcon);

export const SEED_DATA = [
  {
    name: "Row Single Line Body Icon",
    tag: "RowBodyIcon",
    description: "A row with left aligned body text and a right aligned icon",
    category: COMPONENT_TYPES.row,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_SingleLine_BodyIcon.png",
    supports_list_render: true,
    props: {
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      icon: {
        label: "Icon",
        description: "Icon to display",
        type: FORM_TYPES.icon,
        value: "favorite",
        editable: true
      }
    },
    layout: {
      width: 375,
      height: 59
    }
  },
  {
    name: "Row Double Line Body Icon",
    tag: "RowBodyIcon",
    description:
      "A row with left aligned body text and subtitle text and a right aligned icon",
    category: COMPONENT_TYPES.row,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_DoubleLine_BodyIcon.png",
    supports_list_render: true,
    props: {
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      subtitle: {
        label: "Subtitle",
        description: "Subtitle text to display",
        type: FORM_TYPES.string,
        value: "San Diego",
        editable: true
      },
      icon: {
        label: "Icon",
        description: "Icon to display",
        type: FORM_TYPES.icon,
        value: "favorite",
        editable: true
      }
    },
    layout: {
      width: 375,
      height: 81
    }
  }
];
