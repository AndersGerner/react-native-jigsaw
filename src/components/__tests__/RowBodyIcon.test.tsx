import * as React from "react";
import renderer from "react-test-renderer";
import RowBodyIcon from "../RowBodyIcon";

it("renders divider", () => {
  const tree = renderer.create(<RowBodyIcon icon="arrow" />).toJSON();

  expect(tree).toMatchSnapshot();
});
