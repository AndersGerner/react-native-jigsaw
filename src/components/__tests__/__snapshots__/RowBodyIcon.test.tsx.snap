// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders divider 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 26,
            },
            Object {
              "color": "rgba(70, 78, 88, 1)",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
